app/main.py

```diff
@@ -1,3 +1,19 @@
-def cache(func):
-    # Write your code here
-    pass
+import typing
+
+
+def cache(func: typing.Callable) -> typing.Callable:
+    cache_func = {}
+
+    def inner(*args) -> typing.Any:
+
+        if ((args), (func.__name__)) in cache_func:
+            print("Getting from cache")
+
+        else:
+            print("Calculating new result")
+            complited_func = func(*args)
+            key_for_func = ((args), (func.__name__))
+            cache_func[key_for_func] = complited_func
+        return cache_func[(args), (func.__name__)]
```

Comment:
to much brackets here ((args), (func.__name__)), u can write (args, func.__name__) for same result